typescript/packages/subsurface-viewer/src/layers/map/mapLayer.stories.tsx

```diff
@@ -871,6 +873,111 @@ NodeCenteredPropMapWithArrayInput.parameters = {
     },
 };
 
+function makeGaussian(amplitude, x0, y0, stdX, stdY) {
+    return function (amplitude, x0, y0, stdX, stdY, x, y) {
+        const exponent = -(
+            Math.pow(x - x0, 2) / (2 * Math.pow(stdX, 2)) +
+            Math.pow(y - y0, 2) / (2 * Math.pow(stdY, 2))
+        );
+        return amplitude * Math.pow(Math.E, exponent);
+    }.bind(null, amplitude, x0, y0, stdX, stdY);
+}
+
+function makeData(n: number, amplitude: number): Float32Array {
+    const X0 = 0;
+    const Y0 = 0;
+    const stdX = 75;
+    const stdY = 50;
+    const f = makeGaussian(amplitude, X0, Y0, stdX, stdY);
+
+    const data = new Float32Array(n * n).map((val, index) => {
+        const x = (index % n) - n / 2;
+        const y = Math.floor(index / n) - n / 2;
+        return f(x, y); // keep + 0.3 * Math.random();
+    });
+
+    return data;
+}
+
+//-- MapLayer with native javascript arrays as input --
+const TypedArrayInputStory: ComponentStory<typeof SubsurfaceViewer> = (args: {
+    dimension: number;
+}) => {
+    const subsurfaceViewerArgs = {
+        id: "map",
+        layers: [
+            new MapLayer({
+                frame: {
+                    origin: [-args.dimension / 2, -args.dimension / 2],
+                    count: [args.dimension, args.dimension],
+                    increment: [1, 1],
+                    rotDeg: 0,
+                },
+                meshData: makeData(args.dimension, 99),
+                propertiesData: makeData(args.dimension, 1),
+                gridLines: false,
+                material: true,
+                ZIncreasingDownwards: false,
+                contours: [0, 5],
+                colorMapFunction: nearestColorMap as colorMapFunctionType,
+            }),
+            new AxesLayer({
+                ZIncreasingDownwards: false,
+                bounds: [
+                    -args.dimension / 2,
+                    -args.dimension / 2,
+                    -10,
+                    args.dimension / 2,
+                    args.dimension / 2,
+                    60,
+                ],
+            }),
+        ],
+        cameraPosition: {
+            rotationOrbit: 45,
+            rotationX: 45,
+            zoom: [-100, -100, -10, 100, 100, 60],
+            target: [0, 0, 0],
+        },
+        views: {
+            layout: [1, 1],
+            viewports: [
+                {
+                    id: "view_1",
+                    show3D: true,
+                },
+            ],
+        },
+    };
+    return <SubsurfaceViewer {...subsurfaceViewerArgs} />;
+};
+
+export const TypedArrayInput: ComponentStory<typeof SubsurfaceViewer> = (
```

Comment:
`ComponentStory` is deprecated in Storybook > v6

Reply:
Ok I changed it to "StoryFn"